File: algorithms/345.py
# https://leetcode.com/problems/reverse-vowels-of-a-string


class Solution(object):
    def reverseVowels(self, s):
        """
        :type s: str
        :rtype: str
        """

        l = list(s)
        i, j = 0, len(s) - 1
        while i < j:
            li = l[i].lower()
            if li != 'a' and li != 'e' and li != 'i' and li != 'o' and li != 'u':
                i += 1
                continue
            lj = l[j].lower()
            if lj != 'a' and lj != 'e' and lj != 'i' and lj != 'o' and lj != 'u':
                j -= 1
                continue
            l[i], l[j] = l[j], l[i]
            i += 1
            j -= 1
        return ''.join(str(x) for x in l)

File: algorithms/345.swift
// https://leetcode.com/problems/reverse-vowels-of-a-string

class Solution {
    func reverseVowels(s: String) -> String {
        let v = ["A", "E", "I", "O", "U", "a", "e", "i", "o", "u"]
        var s_ = Array(s.characters)
        var i = 0
        var j = s_.count-1
        while i < j {
            if !v.contains(String(s_[i])) {
                i += 1
                continue
            }
            if !v.contains(String(s_[j])) {
                j -= 1
                continue
            }
            let t = s_[i]
            s_[i] = s_[j]
            s_[j] = t
            i += 1
            j -= 1
        }
        return String(s_)
    }
}
